python_modules/dagster/dagster/_core/definitions/decorators/decorator_assets_definition_builder.py

```diff
@@ -682,3 +673,22 @@ def _validate_check_specs_target_relevant_asset_keys(
                 f"Invalid asset key {spec.asset_key} in check spec {spec.name}. Must be one of"
                 f" {valid_asset_keys}"
             )
+
+
+def validate_named_ins_subset_of_deps(
+    named_ins_per_key: Mapping[AssetKey, NamedIn],
+    asset_deps_by_key: Mapping[AssetKey, AssetDep],
+) -> None:
+    """Validates that the asset_ins are a subset of the asset_deps. This is a common validation
+    that we need to do in multiple places, so we've factored it out into a helper function.
+    """
+    asset_dep_keys = set(asset_deps_by_key.keys())
+    asset_in_keys = set(named_ins_per_key.keys())
+
+    if asset_in_keys - asset_dep_keys:
+        invalid_asset_in_keys = asset_in_keys - asset_dep_keys
+        raise DagsterInvalidDefinitionError(
+            f"Invalid asset dependencies: `{invalid_asset_in_keys}` specified as asset inputs, but"
+            " are not specified in `internal_asset_deps`. Asset inputs must be associated with an"
```

Comment:
Should this error message be the one from https://github.com/dagster-io/dagster/pull/26457/files#diff-c693ed7d590d8af6edc03557c1d53e2d32cf76b751d829235813d8a1453062dbL395-L398?